app/test/vesting_test.go

```diff
@@ -0,0 +1,542 @@
+	assert.EqualValues(s.T(),
+		testfactory.BaseAccountDefaultBalance,
+		del[0].Balance.Amount.Int64(),
+		"delegation amount must match")
+}
+
+func (s *VestingModuleTestSuite) testClaimDelegationReward(cctx testnode.Context, accName string) {
+	assert.NoError(s.T(), cctx.WaitForNextBlock())
+
+	accAddress := getAddress(accName, cctx.Keyring).String()
+
+	cli := distributiontypes.NewQueryClient(cctx.GRPCClient)
+	resR, err := cli.DelegationTotalRewards(
+		context.Background(),
+		&distributiontypes.QueryDelegationTotalRewardsRequest{
+			DelegatorAddress: accAddress,
+		},
+	)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), resR, "empty staking rewards")
+	assert.NotEmpty(s.T(), resR.Rewards)
+
+	rewardAmount := resR.Rewards[0].Reward.AmountOf(app.BondDenom).RoundInt().Int64()
+	assert.Greater(s.T(), rewardAmount, int64(0), "rewards must be more than zero")
+
+	balancesBefore, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+
+	// minExpectedBalance is used because more tokens may be vested to the
+	// account in the middle of this test
+	minExpectedBalance := balancesBefore.AmountOf(app.BondDenom).Int64() + rewardAmount
+
+	validators, err := GetValidators(cctx.GRPCClient)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), validators, "empty validators set")
+
+	msg := distributiontypes.NewMsgWithdrawDelegatorReward(
+		getAddress(accName, cctx.Keyring),
+		validators[0].GetOperator(),
+	)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	resTx, err := testnode.SignAndBroadcastTx(encCfg, cctx.Context, accName, []sdk.Msg{msg}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+	assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the claim reward TX must succeed")
+
+	// Check if the reward amount in the account
+	balancesAfter, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+
+	assert.GreaterOrEqual(s.T(), balancesAfter.AmountOf(app.BondDenom).Int64(), minExpectedBalance, "Minimum balance after claiming reward")
+}
+
+func AddAccountsToGenesisState(gs map[string]json.RawMessage, accounts ...authtypes.GenesisAccount) (map[string]json.RawMessage, error) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	var authGenState authtypes.GenesisState
+	err := encCfg.Codec.UnmarshalJSON(gs[authtypes.ModuleName], &authGenState)
+	if err != nil {
+		return gs, err
+	}
+
+	pAccs, err := authtypes.PackAccounts(accounts)
+	if err != nil {
+		return gs, err
+	}
+
+	// set the accounts in the genesis state
+	authGenState.Accounts = append(authGenState.Accounts, pAccs...)
+	gs[authtypes.ModuleName] = encCfg.Codec.MustMarshalJSON(&authGenState)
+
+	return gs, nil
+}
+
+func AddBalancesToGenesisState(gs map[string]json.RawMessage, balances ...banktypes.Balance) (map[string]json.RawMessage, error) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
```

Comment:
[not blocking]
typically we don't creating an encoding config each time we call a function. tbh I don't think its really matters that much, so I'm definitely not blocking on it